elastic/kibana
# [Security Solution] Create rule from timeline

1. Introduces ability to create rule from timeline on the timelines page as shown in the gif below
2. Fixes an unreported bug we had with the current "Import query from saved timeline" button. We only got the query from the timeline, not the index pattern or fields.
3. Introduces a new hook to manage importing the timeline query, index pattern, and fields: `useRuleFromTimeline`

![rft2](https://user-images.githubusercontent.com/6935300/196245858-37607049-1642-4f9c-af28-fb52379d6ca3.gif)

### Cypress
Tests that use the feature  "Import query from saved timeline" now have a different index pattern. We now match exactly the index pattern to the timeline index pattern. So: `getIndexPatterns().join('')` becomes `'auditbeat-*'`. Depending on if any alerts have been generated yet, the timeline index pattern could also include `.alerts-security.alerts-default`. Added a method `removeAlertsIndex` to check if alerts index exists and remove it if it does, or it will break tests.

### How to test

1. Create the following index via dev tools:

```
PUT ho_ho_ho
{
  "mappings": {
    "properties": {
      "good.job": {
        "type": "keyword"
      },
      "@timestamp": {
        "type": "date"
      }
    }
  }
}

POST ho_ho_ho/_doc
{
  "@timestamp": "2022-12-05T06:26:09.015Z",
  "good": { "job": "yessir" }
}
```

2. Create a Data View, `ho_ho_ho`
3. Create 2 saved timelines. One that uses the security data view and queries `host.name: *`, the other one uses the `ho_ho_ho` data view and queries `good.job: *`.
4. From the Timelines page, open the actions menu for the `ho_ho_ho` timeline and select "Create rule from timeline"
5. Once redirected to the rules page, ensure the index pattern and query match. Then use the KQL bar to ensure it is auto suggesting the custom field
6. Use the "Import query from saved timeline" button and select the other timeline with the security data view
7. Ensure the index pattern and query match. Then use the KQL bar to ensure it is auto suggesting our standard ecs fields
